benefits/enrollment_switchio/views.py

```diff
@@ -26,9 +30,43 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class GatewayUrlView(EligibleSessionRequiredMixin, View):
+class GatewayUrlView(AgencySessionRequiredMixin, EligibleSessionRequiredMixin, View):
     """View for the tokenization gateway registration"""
 
-    def get(self, request, *args, **kwargs):
-        data = {"gateway_url": "https://server/gateway/uuid"}
+    def get(self, request: HttpRequest, *args, **kwargs):
+        agency = session.agency(request)
+        switchio_config = agency.switchio_config
+
+        client = Client(
+            api_url=switchio_config.api_base_url,
+            api_key=switchio_config.api_key,
+            api_secret=switchio_config.api_secret,
+            private_key=switchio_config.private_key,
+            client_certificate=switchio_config.client_certificate,
+            ca_certificate=switchio_config.ca_certificate,
+        )
+
+        route = reverse(routes.ENROLLMENT_SWITCHIO_INDEX)
+        redirect_url = _generate_redirect_uri(request, route)
+
+        registration = client.request_registration(
+            eshopRedirectUrl=redirect_url,
+            mode="register",
+            eshopResponseMode=EshopResponseMode.FORM_POST,
+            timeout=settings.REQUESTS_TIMEOUT,
+        )
+
+        data = {"gateway_url": registration.gtwUrl}
         return JsonResponse(data)
+
+
+# copied from https://github.com/Office-of-Digital-Services/django-cdt-identity/blob/main/cdt_identity/views.py#L42-L50
+def _generate_redirect_uri(request: HttpRequest, redirect_path: str):
+    redirect_uri = str(request.build_absolute_uri(redirect_path)).lower()
+
+    # this is a temporary hack to ensure redirect URIs are HTTPS when the app is deployed
+    # see https://github.com/cal-itp/benefits/issues/442 for more context
+    if not redirect_uri.startswith("http://localhost"):
+        redirect_uri = redirect_uri.replace("http://", "https://")
+
+    return redirect_uri
```

Comment:
Just put this here for now... it's being used on [line 50](https://github.com/cal-itp/benefits/pull/2952/files#diff-d6e9f70aaccdb9bc5d303838b7a9481105acb8aca883c47580f46f25a489ca3aR50) to generate the `eshopRedirectUri`. 

Options for what we can do with it are:
- leave it copy-pasted like this
- import the function from `cdt_identity.views` (seems like a weird place to be importing from)
- move the function somewhere and then update `cdt_identity.views` to use it and also use it in `GatewayUrlView.get`.
- leave it duplicated across Benefits and `cdt_identity` but at least put it in some more common module in Benefits rather than here
- something else? idk

Reply:
Let's just leave it copy-pasted so that we don't block this PR. We can discuss this at a future dev workshop